openstudiocore/src/osversion/VersionTranslator.cpp

```diff
@@ -445,6 +446,42 @@ void VersionTranslator::initializeMap(std::istream& is) {
   IddFileAndFactoryWrapper iddFile = getIddFile(currentVersion);
   if (iddFile.iddFileType() == IddFileType::UserCustom) {
     oIdfFile = IdfFile::load(is,iddFile.iddFile());
+    if( currentVersion == VersionString(1,9,0) ) {
+      if( oIdfFile ) {
+        auto sizingObjects = oIdfFile->getObjectsByType(iddFile.getObject("OS:Sizing:Zone").get());
+
+        // Figure out if the OS:Sizing:Zone object looks like it is from CBECC
+        // it will have extra fields that would were not in the "real" 1.9.0 IDD
+        bool fromCBECC = false;
+        for( auto const & object : sizingObjects ) {
+          if( auto value = object.getString(2) ) {
+            if( istringEqual("SupplyAirTemperature", value.get()) ||
+                istringEqual("TemperatureDifference", value.get()) ) {
+              fromCBECC = true;
+              break;
+            }
+          }
+        }
+
+        if( fromCBECC ) {
+          // Remove the sizing objects so that the translation will proceed smoothly
+          oIdfFile->removeObjects(sizingObjects);
+
+          // Get a special CBECC idd file,
+          // load the idf file again against that idd,
+          // get the sizing objects and save them for later,
+          // we will reintrodce the sizing objects in the version 1.10.2 phase of the translation
+          // when they were officially part of OS
+          auto cbeccIddFile = IddFile::load( getSharedResourcesPath() / "osversion/1_9_0_CBECC/OpenStudio.idd");
```

Comment:
I wanted to do that. The trouble is that they are truncated at the max fields of the idd.